lindi/conversion/h5_ref_to_zarr_attr.py

```diff
@@ -0,0 +1,69 @@
+import h5py
+
+
+def h5_ref_to_zarr_attr(ref: h5py.Reference, *, h5f: h5py.File):
+    """Convert/encode an h5py reference to a format that zarr can accept.
+
+    Parameters
+    ----------
+    ref : h5py.Reference
+        The reference to convert.
+    h5f : h5py.File
+        The file that the reference is in.
+
+    Returns
+    -------
+    dict
+        The reference in a format that zarr can accept.
+
+    The format is a dictionary with a single key, '_REFERENCE', whose value is
+    another dictionary with the following keys:
+
+    'object_id', 'path', 'source', 'source_object_id'
+
+    * object_id is the object ID of the target object.
+    * path is the path of the target object.
+    * source is always '.', meaning that path is relative to the root of the
+      file (I think)
+    * source_object_id is the object ID of the source object.
+
+    See
+    https://hdmf-zarr.readthedocs.io/en/latest/storage.html#storing-object-references-in-attributes
+
+    Note that we will also need to handle "region" references. I would propose
+    another field in the value containing the region info. See
+    https://hdmf-zarr.readthedocs.io/en/latest/storage.html#sec-zarr-storage-references-region
+    """
+    file_id = h5f.id
+
+    # The get_name call can actually be quite slow. A possible way around this
+    # is to do an initial pass through the file and build a map of object IDs to
+    # paths. This would need to happen elsewhere in the code.
+    deref_objname = h5py.h5r.get_name(ref, file_id)
+    if deref_objname is None:
+        raise ValueError(f"Could not dereference object with reference {ref}")
+    deref_objname = deref_objname.decode("utf-8")
+
+    dref_obj = h5f[deref_objname]
```

Comment:
Do we need to do this, or could we just do `h5f[ref]`? Or is the idea that this code is what h5py does under the hood, we can make it more efficient by later caching `get_name` calls, and exposing this code now makes it easier to optimize later?